lms/djangoapps/badges/api/tests.py

```diff
@@ -22,6 +22,13 @@ class UserAssertionTestCase(UrlResetMixin, ModuleStoreTestCase, ApiTestCase):
     """
     Mixin for badge API tests.
     """
+    URLS_AUTO_RESET = False
+
+    @classmethod
+    def setUpClass(cls):
```

Comment:
To go further on that thought, if we had that middleware running and shifted over all optional URLs to using it, then we'd never have to worry about resetting URLs because the configuration of the URLs would always match to whatever settings were active at the moment instead of a separate piece of global module state. So the whole of the need for a `UrlResetMixin` class goes away, and we're just left doing what we always do for overriding settings.